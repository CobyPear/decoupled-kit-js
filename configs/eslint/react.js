/** @type {import('eslint').ESLint.ConfigData} */
module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
		'plugin:react/jsx-runtime',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
