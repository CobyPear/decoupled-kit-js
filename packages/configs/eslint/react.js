module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
