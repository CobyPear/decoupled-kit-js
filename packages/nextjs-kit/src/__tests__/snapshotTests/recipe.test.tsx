import { render } from '@testing-library/react';
import Recipe from '../../components/recipe';
import exampleRecipeData from '../data/exampleRecipeData.json';
import { vi } from 'vitest';

vi.mock('../../__mocks__/next/image');

vi.mock('next/router', () => ({
	useRouter: () => ({
		locale: 'en',
	}),
}));

/**
 * @vitest-environment jsdom
 */

describe('<Recipe />', () => {
	it("should render 'recipe'", () => {
		const { asFragment } = render(
			<Recipe
				title={exampleRecipeData[0].title}
				category={exampleRecipeData[0].category}
				imageProps={{
					priority: true,
					src: exampleRecipeData[0].imgSrc,
					layout: 'fill',
					objectFit: 'cover',
				}}
				ingredients={exampleRecipeData[0].ingredients}
				instructions={exampleRecipeData[0].instructions}
			/>,
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
