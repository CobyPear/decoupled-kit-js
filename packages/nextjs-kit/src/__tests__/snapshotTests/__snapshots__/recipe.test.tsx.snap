// Vitest Snapshot v1

exports[`<Recipe /> > should render 'recipe' 1`] = `
<DocumentFragment>
  <article
    class="ps-prose lg:ps-prose-xl ps-mt-10 ps-mx-auto h-fit ps-p-4 sm:ps-p-0"
  >
    <header>
      <h1>
        Guacamole
      </h1>
      <div
        class="ps-flex ps-flex-row ps-justify-between"
      >
        <a
          class="ps-font-normal ps-cursor-pointer"
        >
          Back â†’
        </a>
        <span
          class="text ps-pb-2 ps-pr-3 ps-text-sm text-slate-400"
        >
          Snacks
        </span>
      </div>
    </header>
    <div
      class="ps-relative ps-max-w-lg ps-mx-auto ps-min-w-full ps-h-[50vh] ps-rounded-lg ps-shadow-lg ps-overflow-hidden ps-mt-12 ps-mb-10"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="A bowl of guacamole"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fdev-decoupled-drupal-qa.pantheonsite.io%2Fsites%2Fdefault%2Ffiles%2F2022-07%2Fguacamole.jpeg&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </span>
    </div>
    <div
      class="ps-flex ps-flex-col sm:ps-flex-row"
    >
      <section
        class="ps-flex ps-flex-col min-w-fit sm:ps-border-r-2 ps-pr-4"
      >
        <h2>
          Ingredients
        </h2>
        <ul>
          <li>
            1 Avocado
          </li>
          <li>
            1/2 tomato
          </li>
          <li>
            1/2 onion
          </li>
          <li>
            salt
          </li>
          <li>
            lime
          </li>
          <li>
            black pepper
          </li>
          <li>
            cilantro
          </li>
        </ul>
      </section>
      <section
        class="ps-flex ps-flex-col ps-pl-4"
      >
        <h2
          class="ps-ml-4"
        >
          Directions
        </h2>
        <div>
          <ol>
            
	
            <li>
              Dice the onion and tomato
            </li>
            
	
            <li>
              Scoop the avocado insides into a bowl
            </li>
            
	
            <li>
              Add the tomato, onion, cilantro, lime juice, salt, and pepper
            </li>
            
	
            <li>
              Mix together
            </li>
            
	
            <li>
              Serve
            </li>
            

          </ol>
          

        </div>
      </section>
    </div>
  </article>
</DocumentFragment>
`;
